frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -1061,21 +1149,123 @@ mod tests {
 			assert!(MultiPhase::current_phase().is_unsigned());
 
 			// first execution -- okay.
-			assert!(MultiPhase::try_acquire_offchain_lock(25).is_ok());
+			assert!(MultiPhase::ensure_offchain_repeat_frequency(25).is_ok());
 
 			// next block: rejected.
-			assert_noop!(MultiPhase::try_acquire_offchain_lock(26), MinerError::Lock("recently executed."));
+			assert_noop!(
+				MultiPhase::ensure_offchain_repeat_frequency(26),
+				MinerError::Lock("recently executed.")
+			);
 
 			// allowed after `OFFCHAIN_REPEAT`
-			assert!(MultiPhase::try_acquire_offchain_lock((26 + offchain_repeat).into()).is_ok());
+			assert!(
+				MultiPhase::ensure_offchain_repeat_frequency((26 + offchain_repeat).into()).is_ok()
+			);
 
 			// a fork like situation: re-execute last 3.
-			assert!(MultiPhase::try_acquire_offchain_lock((26 + offchain_repeat - 3).into()).is_err());
-			assert!(MultiPhase::try_acquire_offchain_lock((26 + offchain_repeat - 2).into()).is_err());
-			assert!(MultiPhase::try_acquire_offchain_lock((26 + offchain_repeat - 1).into()).is_err());
+			assert!(MultiPhase::ensure_offchain_repeat_frequency(
+				(26 + offchain_repeat - 3).into()
+			)
+			.is_err());
+			assert!(MultiPhase::ensure_offchain_repeat_frequency(
+				(26 + offchain_repeat - 2).into()
+			)
+			.is_err());
+			assert!(MultiPhase::ensure_offchain_repeat_frequency(
+				(26 + offchain_repeat - 1).into()
+			)
+			.is_err());
 		})
 	}
 
+	#[test]
+	fn ocw_lock_prevents_overlapping_execution() {
+		{
+			// first, ensure that a successful execution releases the lock
+			let (mut ext, pool) = ExtBuilder::default().build_offchainify(0);
+			ext.execute_with(|| {
+				let running = StorageValueRef::persistent(&OFFCHAIN_RUNNING);
+				let last_block = StorageValueRef::persistent(OFFCHAIN_LAST_BLOCK);
+
+				roll_to(25);
+				assert!(MultiPhase::current_phase().is_unsigned());
+
+				// initially, the running flag is not set.
+				assert!(matches!(running.get::<bool>(), None));
+				assert!(matches!(last_block.get::<BlockNumber>(), None));
+
+				// a successful a-z execution.
+				MultiPhase::offchain_worker(25);
+				assert_eq!(pool.read().transactions.len(), 1);
+
+				// afterwards, it is flipped to false.
+				assert!(matches!(running.get::<bool>(), Some(Some(false))));
+				assert!(matches!(last_block.get::<BlockNumber>(), Some(Some(25))));
+			});
+		}
+
+		{
+			// ensure that if the running flag is set, a new execution is not allowed.
+			let (mut ext, _) = ExtBuilder::default().build_offchainify(0);
+			ext.execute_with(|| {
```

Comment:
Each of these scopes could be their own test. The benefit is that they can fail independently, which gives more information in case one of them does start to fail.

Reply:
for now I removed these due to rewrite, will try and add some new tests.